php_phongo.c

```diff
@@ -456,180 +467,179 @@ mongoc_bulk_operation_t *phongo_bulkwrite_init(zend_bool ordered) { /* {{{ */
 	return mongoc_bulk_operation_new(ordered);
 } /* }}} */
 
-static bool process_read_concern(zval *option, bson_t *mongoc_opts TSRMLS_DC)
+/* Parses the "readConcern" option for an execute method. If mongoc_opts is not
+ * NULL, the option will be appended. On error, false is returned and an
+ * exception is thrown. */
+static bool phongo_parse_read_concern(zval *options, bson_t *mongoc_opts TSRMLS_DC) /* {{{ */
 {
-	if (Z_TYPE_P(option) == IS_OBJECT && instanceof_function(Z_OBJCE_P(option), php_phongo_readconcern_ce TSRMLS_CC)) {
-		const mongoc_read_concern_t *read_concern = phongo_read_concern_from_zval(option TSRMLS_CC);
+	zval *option = NULL;
+	mongoc_read_concern_t *read_concern;
 
-		if (!mongoc_read_concern_append((mongoc_read_concern_t*)read_concern, mongoc_opts)) {
-			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Error appending \"%s\" option", "readConcern");
-			return false;
-		}
-	} else {
-		phongo_throw_exception(
-			PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC,
-			"Expected 'readConcern' option to be 'MongoDB\\Driver\\ReadConcern', %s given",
-			zend_get_type_by_const(Z_TYPE_P(option))
-		);
+	if (!options) {
+		return true;
+	}
+
+	if (Z_TYPE_P(options) != IS_ARRAY) {
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected options to be array, %s given", zend_get_type_by_const(Z_TYPE_P(options)));
```

Comment:
Should use the new macro to show object name.

Reply:
I implemented https://github.com/mongodb/mongo-php-driver/pull/705 independently and wasn't about to rebase this atop it. I do intend to update all of these to use the new macros after #705 is merged and I rebase, though.